File: Java/DesignPatterns/Singleton.java
// Singleton design pattern in Java
public class Singleton
{
    // The private instance of the singleton class:
    private static Singleton instance;

    // Private data associated with the singleton:
    bool logEnabled;

    // Private constructor:
    private Singleton()
    {
        logEnabled = true;
    }

    // Private instance getter:
    private static Singleton GetInstance()
    {
        if(instance == null)
        {
            instance = new Singleton();
        }
        return instance;
    }

    // Public static methods for accessing the singleton class:

    // For this example, we'll use it as a logging service:
    public static void EnableLogging(bool state)
    {
        // Get the instance via the static method:
        Singleton logger = Singleton.GetInstance();
        // Set the instance state:
        logger.logEnabled = state;
    }

    public static void LogMessage(string message)
    {
        // Get the instance via the static method:
        Singleton logger = Singleton.GetInstance();
        // Check the instance's logging state:
        if(logger.logEnabled)
        {
            // Write to the console window:
            System.out.println(message);
        }
    }
}

File: PHP/DesignPatterns/Singleton.php
<?php

    class Singleton
    {
        // The static instance:
        private static $instance;
        // Instance data:
        private $logEnabled;

        // Private construction:
        private final function __construct()
        {
            $this->logEnabled = false;
        }

        // Private instance getter:
        private static function GetInstance()
        {
            if(!isset(self::$instance))
            {
                self::$instance = new Singleton();
            }
            return self::$instance;
        }

        // Public static methods for accessing the singleton class:

        // In this example, we'll be implementing a basic logging system:
        public static function EnableLogging($state)
        {
            $logger = self::GetInstance();

            $logger->logEnabled = $state;
        }

        public static function LogMessage($message)
        {
            $logger = self::GetInstance();

            if($logger->logEnabled == true)
            {
                echo $message;
            }
        }
    }